### PR TITLE
WhereAmI: Map template added

### PR DESCRIPTION
**Pull request by amoraleda**
**Fixes** #445

Added  `structured_answer` to show a map instead of a text.

TO_FIX: It seems to have some sort of problem in the front-end and doesn't render the map.

---
IA:[https://duck.co/ia/view/where_am_i]( https://duck.co/ia/view/where_am_i)

### CODE REVIEW

**Comment by daxtheduck** [comment]
@amoraleda Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/WhereAmI.pm [hunk 1]
```diff
@@ -9,17 +9,33 @@ zci is_cached => 0;
...
+    
     return $answer,
-      structured_answer => {
-        input     => [],
-        operation => 'Apparent current location',
-        result    => $answer
-      };
+        structured_answer => {
+            id => 'whereami',
+            name => 'Map',
```
Comment (moollaza) [inline lib/DDG/Goodie/WhereAmI.pm]: I would go with `Answer` for the tab name

File: lib/DDG/Goodie/WhereAmI.pm [hunk 2]
```diff
@@ -9,17 +9,33 @@ zci is_cached => 0;
...
+            data => {
+                id => 'whereami_id',
+                name => 'Apparent current location',
+                city => $loc->city,
+                lat => $loc->latitude,
+                lon => $loc->longitude
+            },
+            meta => {
+                zoomValue => 3,
+                sourceName => 'DDG GeoIP',
```
Comment (moollaza) [inline lib/DDG/Goodie/WhereAmI.pm]: We can remove the sourceName/Url

File: lib/DDG/Goodie/WhereAmI.pm [hunk 3]
```diff
@@ -5,21 +5,29 @@ use strict;
...
 
-    return $answer,
-      structured_answer => {
-        input     => [],
-        operation => 'Apparent current location',
-        result    => $answer
-      };
+    return "",
+        structured_answer => {
+            id => 'where_am_i',
```
Comment (moollaza) [inline lib/DDG/Goodie/WhereAmI.pm]: We've now changed Goodies to no longer use the `id` or `name` in the Perl -- let's remove it from here

**Comment by moollaza** [comment]
@amoraleda Thanks! This LGTM :+1:

**Pull request merged**